### DISCUSSION TITLE
Tasmota with MODBus device PowMr solar inverter

### DISCUSSION THREAD

**Discussion by Mape24** [arendst/Tasmota#22364] [Show and tell]
Hi, I have managed to get my PowMr solar inverter connected to Home Assistant by means of Tasmota. If you are interested - I have collected the details here:
https://petschke.familyds.net/dokuwiki/doku.php?id=technik:pow-hvm6.2m_with_tasmota#tasmota_modbus_script

**Comment by pkkrusty** [comment]
Nice, I was just working on a script for PowMr as well. Because the bit rate is so low, you would be better served pulling all the registers at once using r 050311960036 and then pulling values out of that single transaction. 

You could also build with the JSON option and add a >J section to publish via mqtt, and integrate into home assistant that way. That way you can include whatever additional info you want (units)

Check out the example for the Huawei 10ktl on the SML documentation page for an example

**Reply by Mape24** [reply]
Hi Pkkrusty, 
yes **polling at once** is a nice idea. I didn't get it working at the (my) first trial. Please feel free to continue at this point.
I used **MQTT** for some values and indeed - the MQTT integration works a bit better at this point because you can define **unit_of_measurement** in the HA _configuration.yaml_ which enables a nicer history diagram at Home Assistant.
